src/util/yp_newline_list.c

```diff
@@ -32,9 +32,10 @@ yp_newline_list_append(yp_newline_list_t *list, const char *cursor) {
 
     assert(cursor >= list->start);
     size_t newline_offset = (size_t) (cursor - list->start + 1);
-    assert(list->size == 0 || newline_offset > list->offsets[list->size - 1]);
-    list->offsets[list->size++] = newline_offset;
 
+    if (list->size == 0 || newline_offset > list->offsets[list->size - 1]) {
```

Comment:
I think we want to keep around the assertion. It looks like we're probably calling this in a place we shouldn't be.